File: Java/Factorization Of Numbers/PrimeFactorization.java
public class PrimeFactorization {
	// Function to generate factor by multipling current prime factor to all element of list
	private static int generate_factors(int[] l, int ptr, int pf, int con) {
		for (int i=ptr-1; i>=0; i--)
			// As l[0] = 1, so if n*1 will n, and we want only one n in list
			if ((i==0 && con==1) || i!=0)
				l[ptr++] = l[i]*pf;
		return ptr;
	}


	// Function to generate list of factors
	private static int[] get_factorsList(int n) {
		// Initialize array
		int[] list = new int[(int)Math.sqrt(n)+1];
		int ptr = 0;
		// Counter variable, to state iterate no.
		int condition = 1;

		list[ptr++] = 1;	// Add 1, as it is factor of all no.

		// Check if 2 is in prime factor
		while (n%2 == 0) {
			ptr = generate_factors(list, ptr, 2, condition);
			condition = 0;
			n /= 2;
		}

		// Check if other odd prime factors
		for (int i=3; i<=(int)Math.sqrt(n); i+=2) {
			condition = 1;
			// Check if i is in prime factor
			while (n%i == 0) {
				ptr = generate_factors(list, ptr, i, condition);
				condition = 0;
				n /= i;
			}
		}

		// to append last prime factor in list
		if (n > 2) { generate_factors(list, ptr, n, 1); }

		return list;
	}


	public static void main(String[] args) {
		// Input Array
		int[] arr = {23, 46, 65, 34234, 423, 43212};

		for (int i=0; i<arr.length; i++) {
			// Get list factor of arr[i]
			int[] temp = get_factorsList(arr[i]);

			// Print all factors
			for (int j=0; j<temp.length&&temp[j]!=0; j++)
				System.out.print(temp[j]+" ");
			System.out.println();
		}
	}
}

File: Python/Factorization Of Numbers/PrimeFactorization.py
# Function to generate factor by multipling current prime factor to all element of list
def generate_factors(l, n, c):
	L= len(l)
	for i in range(L):
		# As l[0] = 1, so if n*1 will n, and we want only one n in list
		if (i==0 and c) or i!=0:
			l.append(l[i]*n)


# function to generate list of factors
def get_factorList(n):
	l = [1]	# List to store array of factor
	conditon = True	# Counter variable, to state iterate no.

	# Check if 2 is in prime factor
	while n%2 == 0:
		generate_factors(list, 2, conditon)
		conditon = False
		n //= 2

	# Check if other odd prime factors
	for i in range(3, int(n**0.5)+1, 2):
		conditon = True
		# Check if i is in prime factor
		while n%i == 0:
			generate_factors(list, i, conditon)
			conditon = False
			n //= i

	if n > 2:
		# to append last prime factor in list
		generate_factors(list, n, True)

	return list


if __name__ == "__main__":
	# List of input no's
	list_of_numbers = [23, 46, 65, 34234, 423, 43212]

	# Get factor list of given no.
	for num in list_of_numbers:
		print(get_factorList(num))